### PR TITLE
Code Upgrade part 1

### PR DESCRIPTION
**Pull request by codyhoag**
Hey Greg,

This pull request covers the first four steps of the Upgrade Planner's checklist:
1. Upgrading Your Development Environment
2. Migrating Plugins SDK Projects to Liferay Workspace
3. Upgrading Build Dependencies
4. Fixing Upgrade Problems

Do you have time to review this soon for accuracy? I'd like to get this published ASAP so the tool can have some docs to reference. Please let me know any suggestions you have. Thanks!

### CODE REVIEW

**Comment by gamerson** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: developer/tutorials/articles/03-upgrading-code-to-liferay-7.2/01-intro.markdown [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+        1.  Create New Liferay Workspace <!--commandId="create_new_liferay_workspace"-->
+
+            Create a new Liferay Workspace.
+
+        2.  Import Existing Liferay Workspace
+
+            If you do not have an existing 7.x workspace, go to the previous
+            step.
+
+    2.  [Configure Liferay Workspace Settings](/docs/tutorials/7-2/-/knowledge_base/t/upgrading-your-development-environment#configuring-liferay-workspace)
```
Comment (gamerson) [inline developer/tutorials/articles/03-upgrading-code-to-liferay-7.2/01-intro.markdown:25]: Where do these links point to?  do these pages exist anywhere?

Comment (codyhoag) [reply developer/tutorials/articles/03-upgrading-code-to-liferay-7.2/01-intro.markdown:25]: This specific one points [here](https://github.com/gamerson/liferay-docs/blob/86fbc3be209ae32785c23eeb5101b9fb6c18ac24/developer/tutorials/articles/03-upgrading-code-to-liferay-7.2/02-upgrading-your-development-environment.markdown#configuring-liferay-workspace). These are links that point to docs in this PR. These will be changing though to match exact steps in tool. But these should provide good examples for testing.

**Comment by jtydhr88** [comment]
hey @gamerson @codyhoag 
As Greg mentioned that we enable custom attributes support, I'd like to add the change https://github.com/codyhoag/liferay-docs/pull/153 to allow Upgrade Planner to get commondId from attributes.

**Comment by codyhoag** [comment]
Thanks @jtydhr88! I'll definitely add those in. I'm hoping to wrap up the initial draft for the code upgrade docs by EOD tomorrow (5/21). I'll send for SME review to make sure everything looks OK.

I also saw @gamerson was adding in `.summary` and `.title` attributes when I talked to him last, and there were no single quotes around the command IDs:
```
1.  [Upgrade Your Development Environment](/docs/tutorials/7-2/-/knowledge_base/t/upgrading-your-development-environment){.title}

    Legacy project environments should be upgraded to the latest version of
    Liferay Workspace to ensure you leverage all available features.{.summary}

    1.  Set up Liferay Workspace{.title}

        1.  Create New Liferay Workspace{.title, commandId=create_new_liferay_workspace}

            If you already have an existing workspace, you may skip this step
            otherwise select Perform to create a new Liferay workspace.{.summary}

        2.  Import Existing Liferay Workspace{.title, commandId=import_existing_liferay_workspace}

            If you do not have an existing 7.x workspace, go to the previous
            step.{.summary}
```
Is that the way we want to go? Just let me know. Thanks!

**Comment by jtydhr88** [comment]
hey @codyhoag please verify the structure with @gamerson 
and I also need https://github.com/codyhoag/liferay-docs/pull/154 one commit for upgrade planner, it will allow upgrade planner to get required properties.
@gamerson I also sent IDE PR about it https://github.com/gamerson/liferay-ide/pull/4464

**Comment by codyhoag** [comment]
Wrapping up the final draft for the Code Upgrade docs. Will send in new PR for SME review

**Comment by codyhoag** [comment]
Resent final drafts gamerson#36

**Pull request closed without merging**